js/coinsbit.js

```diff
@@ -0,0 +1,1070 @@
+            result.push (this.parseTicker (tickers[marketId], market));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //       at: '1640028811',
+        //       ticker: {
+        //         bid: '46448.49682539',
+        //         ask: '46458.41093569',
+        //         open: '46687.39',
+        //         high: '47550',
+        //         low: '45584.27',
+        //         last: '46462.77',
+        //         vol: '32.56275846',
+        //         deal: '1510574.62551143',
+        //         change: '-0'
+        //       }
+        let tickerFinal = {};
+        let timestamp = undefined;
+        if ('ticker' in ticker) { // if arrived from fetchTickers
+            tickerFinal = this.safeValue (ticker, 'ticker', {});
+            timestamp = this.safeTimestamp (ticker, 'at');
+        } else { // if arrived from fetchTicker, then there is nothing to be done more
+            tickerFinal = ticker;
+            timestamp = this.milliseconds ();
+        }
+        const last = this.safeNumber (tickerFinal, 'last');
+        market = this.safeMarket (undefined, market);
+        return this.safeTicker ({
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeNumber (tickerFinal, 'high'),
+            'low': this.safeNumber (tickerFinal, 'low'),
+            'bid': this.safeNumber (tickerFinal, 'bid'),
+            'bidVolume': undefined,
+            'ask': this.safeNumber (tickerFinal, 'ask'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': this.safeNumber (tickerFinal, 'open'),
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined, // They do have 'change' property, but atm,  for some reason, they always return "0 | -0 | -1 | -2" strange values. So, skipping for now.
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeNumber (tickerFinal, 'volume'),
+            'quoteVolume': this.safeNumber (tickerFinal, 'deal'),
+            'info': ticker,
+        }, market);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = Math.min (1000, limit); // this parameter is not required, default 50 will be returned
+        }
+        const type = this.safeString (params, 'type', 'depth/result'); // <<<TODO>> maybe a better way to set default unified param name (same param can be used for binance fetchBidsAsks too, if that will get integrated inside binance's fetchOrderbook )
```

Comment:
I suggest that we don't use the name `type` for this purpose, since `type` usually refers to market types, like spot, swap, futures, etc.  

Another thing here – let's not parameterize it with the pathname like `depth/result`. 

We parameterize it with implicit method names instead. Hence the name of this parameter is `method` instead of `type`.

```JavaScript
const options = this.safeValue (this.options, 'fetchOrderBook', {});
// publicGetDepthResult or publicGetBook
const type = this.safeString (params, 'method', 'publicGetDepthResult');
const method = this.safeString (options, 'method', type);
params = this.omit (params, 'method'); // don't forget to clean it up
if (method === 'publicGetBook') {
    let query = this.extend (request, { 'side': 'buy' });
    const requestBids = await this[method](this.extend (query, params));
    query = this.extend (request, { 'side': 'sell' });
    const requestAsks = await this[method] (this.extend (query, params));
    //
    //     ...
    //
} else {
    response = await this[method] (this.extend (request, params));
    //
    //     {
    //         asks: [
    //             [  "46809.06069986", "2.651"  ],
    //             [  "46819.80796146", "2.6401" ],
    //             ...
    //         ],
    //         bids: [
    //             [  "46799.39054516",  "2.702" ],
    //             [  "46801.25196599",  "0.4"   ],
    //             ...
    //         ]
    //     }
    //
    priceKey = 0;
    amountKey = 1;
} 
```